source/configuration_manual/authentication/oauth2.rst

```diff
@@ -253,60 +266,88 @@ Oauth2 overrides some of the default HTTP client and SSL settings:
+   :seealso: @dict
+   :values: @named_filter
+
+   A dictionary for fetching validation keys.
+
+   Example:
+
+   .. code-block:: none  
+
+     local_validation {
+       dict_driver = fs
+       fs posix {
+       }
+       fs_posix_prefix = /tmp/keys/
+     }
 
-  ## url for verifying token validity. Token is appended to the URL
-  # tokeninfo_url = http://endpoint/oauth/tokeninfo?access_token=
+.. dovecot_core:setting:: oauth2_scope
+   :values: @string
 
-  ## introspection endpoint, used to gather extra fields and other information.
-  # introspection_url = http://endpoint/oauth/me
+   A space separated list of scopes of validity. (optional)
 
-  ## How introspection is made, valid values are
-  ##   auth = GET request with Bearer authentication
-  ##   get  = GET request with token appended to URL
-  ##   post = POST request with token=bearer_token as content
-  ##   local = Attempt to locally validate and decode JWT token
-  # introspection_mode = auth
 
-  ## Force introspection even if tokeninfo contains wanted fields
-  ## Set this to yes if you are using active_attribute
-  # force_introspection = no
+.. dovecot_core:setting:: oauth2_username_attribute
+   :values: @string
+   :default: email
 
-  ## Validation key dictionary, turns on local validation
-  # local_validation_key_dict =
+   Username attribute in response.
 
-  ## A space separated list of scopes of validity (optional)
-  # scope = something
+.. dovecot_core:setting:: oauth2_active_attribute
+   :values: @string
 
-  ## username attribute in response (default: email)
-  # username_attribute = email
+   Attribute name for checking whether account is disabled. (optional)
 
-  ## username normalization format (default: %Lu)
-  # username_format = %Lu
+.. dovecot_core:setting:: oauth2_active_value
+   :values: @string
 
-  ## Attribute name for checking whether account is disabled (optional)
-  # active_attribute =
+   Expected value in active_attribute. (empty = require present, but anything goes)
 
-  ## Expected value in active_attribute (empty = require present, but anything goes)
-  # active_value =
+.. dovecot_core:setting:: oauth2_issuers
+   :values: @string
 
-  ## Expected issuer(s) for the token (space separated list)
-  # issuers =
+   Expected issuer(s) for the token. (space separated list)
 
+.. dovecot_core:setting:: oauth2_openid_configuration_url
+   :values: @string
 
-  ## URL to RFC 7628 OpenID Provider Configuration Information schema
-  # openid_configuration_url =
+   URL to :rfc:`7628` OpenID Provider Configuration Information schema.
 
-  ## Extra fields to set in passdb response (in passdb static style)
-  # pass_attrs =
+.. dovecot_core:setting:: oauth2_fields
+   :values: @named_filter
 
-  ## Timeout in milliseconds
-  # timeout_msecs = 0
+   Key-value fields to include in successful authentication.
 
-  ## Enable debug logging
-  # debug = no
+.. dovecot_core:setting:: oauth2_blocking
+   :values: @boolean
```

Comment:
This is done by the mechanism code, not passdb? Anyway for passdb/userdb I renamed this to use_worker to make it clearer, so this should be oauth2_use_worker.